Doctrine/Orm/DataProvider.php

```diff
@@ -118,8 +118,12 @@ public function getCollection(ResourceInterface $resource, Request $request)
             }
         }
 
-        if (null !== $this->order) {
-            $queryBuilder->addOrderBy('o.id', $this->order);
+        $classMetaData = $manager->getClassMetadata($entityClass);
+        $identifierList = $classMetaData->getIdentifier();
+
+        if (null !== $this->order && count($identifierList) == 1) {
+            $identifier = array_shift($identifierList);
```

Comment:
better use `$identifierList[0]` than array shift (faster).

Btw, we generally use plural for collection (`$identifiers`).